packages/devtools_app/lib/src/screens/memory/memory_tabs.dart

```diff
@@ -107,7 +110,7 @@ class _MemoryTabsState extends State<MemoryTabs>
   @override
   Widget build(BuildContext context) {
     final themeData = Theme.of(context);
-
+    final isProfile = serviceManager.connectedApp?.isProfileBuildNow ?? false;
```

Comment:
How about making isProfile a getter, and maybe rename it to shouldShowTrace, so that we do not have logic duplicates?

Reply:
If isProfile accidentally provide different values for both invocations, we will have UI failure, because number for tab headers will not match number of tabs.

Lets calculate it in initTabs and just use the value in build.